src/sentry/event_manager.py

```diff
@@ -284,26 +284,37 @@ def get_stored_crashreports(cache_key: str | None, event: Event, max_crashreport
 
 
 class ScoreClause(Func):
-    def __init__(self, group=None, last_seen=None, times_seen=None, *args, **kwargs):
+    def __init__(
+        self,
+        group: Group | None = None,
+        last_seen: datetime | None = None,
+        times_seen: int | None = None,
+        *args: Any,
+        **kwargs: Any,
+    ):
         self.group = group
         self.last_seen = last_seen
         self.times_seen = times_seen
         # times_seen is likely an F-object that needs the value extracted
         if hasattr(self.times_seen, "rhs"):
-            self.times_seen = self.times_seen.rhs.value
+            self.times_seen = self.times_seen.rhs.value  # type: ignore[union-attr]
         super().__init__(*args, **kwargs)
 
-    def __int__(self):
+    def __int__(self) -> int:
         # Calculate the score manually when coercing to an int.
         # This is used within create_or_update and friends
         return self.group.get_score() if self.group else 0
 
-    def as_sql(self, compiler, connection, function=None, template=None):
+    def as_sql(
+        self,
+        *args: Any,
+        **kwargs: Any,
+    ) -> tuple[str, list[Any]]:
         has_values = self.last_seen is not None and self.times_seen is not None
         if has_values:
-            sql = "log(times_seen + %d) * 600 + %d" % (
+            sql = "log(times_seen + %d) * 600 + %d" % (  # type: ignore[str-format]
                 self.times_seen,
-                self.last_seen.timestamp(),
+                self.last_seen.timestamp(),  # type: ignore[union-attr]
```

Comment:
""

Reply:
@asottile-sentry it seems like mypy isn't smart enough to pick up the fact that `last_seen` cannot be none here because of the check on `has_values`. Do I need ot make actual functional code changes to make it pass typing?

Reply:
yes often you have to make slight changes to make the code more readable (and understandable to the type checker) -- I think the assignment ot `has_values` can just be inlined into the `if` and it will understand